src/Ads/AccountService.php

```diff
@@ -178,6 +180,17 @@ public function setup_account(): array {
 						$this->link_merchant_account();
 						break;
 
+					case 'account_access':
+						if ( ! $this->check_ads_account_has_access() ) {
+							$state[ $name ]['status'] = AdsAccountState::STEP_PENDING;
+							$this->state->update( $state );
+
+							// Exit the setup loop.
+							break 2;
```

Comment:
I've updated this in [957949a](https://github.com/woocommerce/google-listings-and-ads/pull/2270/commits/957949a9d1a79136ce6f14fd5d8ddccbdf9917aa). I was trying to avoid the 428 altogether, but manually setting the step to PENDING and then forcing the loop to stop execution didn't seem elegant either. From what I can tell, the return value from the POST request is not required by the setup process in order to work properly, so this should be ok until #2345 is resolved.